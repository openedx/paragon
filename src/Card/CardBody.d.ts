import React from 'react';

export interface CardBodyProps {
  className?: string,
  children?: React.ReactNode,
}
declare const CardBody: React.ForwardRefExoticComponent<CardBodyProps>;
export default CardBody;
