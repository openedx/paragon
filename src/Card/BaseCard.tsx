import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import type { ComponentWithAsProp, BsPropsWithAs } from '../utils/types/bootstrap';

// @ts-ignore
import CardBody from './CardBody';

const BASE_CARD_CLASSNAME = 'card';

const colorVariants = [
  'primary',
  'secondary',
  'success',
  'danger',
  'warning',
  'info',
  'dark',
  'light',
] as const;

const textVariants = [
  'white',
  'muted',
] as const;

type ColorVariant = typeof colorVariants[number];
type TextVariant = typeof textVariants[number];
interface Props extends BsPropsWithAs {
  prefix?: string;
  bgColor?: ColorVariant;
  textColor?: ColorVariant | TextVariant;
  borderColor?: ColorVariant;
  hasBody?: boolean;
  className?: string;
  children: React.ReactNode;
}
type BaseCardType = ComponentWithAsProp<'div', Props>;

const BaseCard : BaseCardType = React.forwardRef<HTMLDivElement, Props>(
  (
    {
      prefix,
      className,
      bgColor,
      textColor,
      borderColor,
      hasBody = false,
      children,
      as: Component = 'div',
      ...props
    },
    ref,
  ) => {
    const classes = classNames(
      className,
      prefix ? `${prefix}-${BASE_CARD_CLASSNAME}` : BASE_CARD_CLASSNAME,
      bgColor && `bg-${bgColor}`,
      textColor && `text-${textColor}`,
      borderColor && `border-${borderColor}`,
    );

    return (
      <Component ref={ref} {...props} className={classes}>
        {hasBody ? <CardBody>{children}</CardBody> : children}
      </Component>
    );
  },
);

/* eslint-disable react/require-default-props */
BaseCard.propTypes = {
  /** Prefix for component CSS classes. */
  prefix: PropTypes.string,
  /** Background color of the card. */
  bgColor: PropTypes.oneOf(colorVariants),
  /** Text color of the card. */
  textColor: PropTypes.oneOf([...colorVariants, ...textVariants]),
  /** Border color of the card. */
  borderColor: PropTypes.oneOf(colorVariants),
  /** Determines whether the card should render its children inside a `CardBody` wrapper. */
  hasBody: PropTypes.bool,
  /** Set a custom element for this component. */
  as: PropTypes.elementType,
  /** Additional CSS class names to apply to the card element. */
  className: PropTypes.string,
  /** The content to render inside the card. */
  children: PropTypes.node,
};

export default BaseCard;
