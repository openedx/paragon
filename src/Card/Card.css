/**
 * Do not edit directly
 * Generated on Wed, 01 Mar 2023 00:27:19 GMT
 */

:root {
  --pgn-spacing-card-focus-border-offset: 5px;
  --pgn-spacing-card-logo-bottom-offset-horizontal: .4375rem;
  --pgn-spacing-card-logo-bottom-offset-base: 1rem;
  --pgn-spacing-card-logo-left-offset-horizontal: .4375rem;
  --pgn-spacing-card-logo-left-offset-base: 1.5rem;
  --pgn-spacing-card-loading-skeleton-spacer: .313rem;
  --pgn-spacing-card-footer-action-gap: .5rem;
  --pgn-spacing-card-columns-gap: 1.25rem;
  --pgn-spacing-card-columns-count: 3;
  --pgn-spacing-card-margin-group: 12px;
  --pgn-spacing-card-spacer-y: .75rem;
  --pgn-spacing-card-spacer-x: 1.25rem;
  --pgn-size-card-logo-height: 4.125rem;
  --pgn-size-card-logo-width: 7.25rem;
  --pgn-size-card-image-vertical-max-height: 140px;
  --pgn-size-card-image-horizontal-width-max: 240px;
  --pgn-size-card-focus-border-radius: 5px;
  --pgn-size-card-focus-border-width: 2px;
  --pgn-size-card-border-radius-logo: .25rem;
  --pgn-size-card-border-radius-image: .3125rem;
  --pgn-typography-footer-text-font-size: var(--pgn-typography-font-size-small-x);
  --pgn-spacing-card-columns-margin: var(--pgn-spacing-card-spacer-y);
  --pgn-spacing-card-margin-grid: var(--pgn-spacing-card-margin-group);
  --pgn-spacing-card-margin-deck: var(--pgn-spacing-card-margin-group);
  --pgn-size-card-image-horizontal-width-min: var(--pgn-size-card-image-horizontal-width-max);
  --pgn-size-card-border-radius-base: var(--pgn-size-border-radius-base);
  --pgn-size-card-border-width: var(--pgn-size-border-width);
  --pgn-spacing-card-margin-grid-bottom: var(--pgn-spacing-spacer-3);
  --pgn-spacing-card-margin-deck-bottom: var(--pgn-spacing-spacer-3);
}
