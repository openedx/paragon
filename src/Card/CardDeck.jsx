import React, { Children, useMemo } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import BaseCardDeck from 'react-bootstrap/CardDeck';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { useOverflowScrollItems } from '../OverflowScroll';

const CardDeck = React.forwardRef(({
  className,
  children,
  columnSizes,
  hasInteractiveChildren,
  canScrollHorizontal,
  hasOverflowScrollItems,
  hasEqualColumnHeights,
}, ref) => {
  const cards = useMemo(
    () => Children.map(children, (card) => (
      <Col
        {...columnSizes}
        className={classNames(
          'pgn__card-deck__card-item',
          {
            'pgn__card__disable-equal-column-heights': !hasEqualColumnHeights,
          },
        )}
      >
        {card}
      </Col>
    )),
    [children, columnSizes, hasEqualColumnHeights],
  );
  const overflowCardDeckItems = useOverflowScrollItems(cards);

  const cardDeckChildren = useMemo(() => {
    if (hasOverflowScrollItems) {
      return overflowCardDeckItems;
    }
    return cards;
  }, [hasOverflowScrollItems, overflowCardDeckItems, cards]);

  return (
    <div
      className={classNames(
        'pgn__card-deck',
        {
          'pgn__card-deck-has-horizontal-scroll': canScrollHorizontal,
        },
        className,
      )}
    >
      <Row
        className="pgn__card-deck-row"
        tabIndex={hasInteractiveChildren ? -1 : 0}
        ref={ref}
      >
        {cardDeckChildren}
      </Row>
    </div>
  );
});

CardDeck.propTypes = {
  /** The class name for the CardDeck component */
  className: PropTypes.string,
  /** The Card components to organize */
  children: PropTypes.node.isRequired,
  /**
   * An object containing the desired column size at each breakpoint, following a similar
   * props API as ``react-bootstrap/Col``
   */
  columnSizes: PropTypes.shape({
    xs: PropTypes.number,
    sm: PropTypes.number,
    md: PropTypes.number,
    lg: PropTypes.number,
    xl: PropTypes.number,
  }),
  /** Whether the child `Card` components are interactive/focusable. If not, a `tabindex="0"` is
   * added to be a11y-compliant */
  hasInteractiveChildren: PropTypes.bool,
  /** Whether the `CardDeck` supports horizontal scrolling when there are overflow children */
  canScrollHorizontal: PropTypes.bool,
  /** Whether the children of CardDeck should be processed by `useOverflowScrollItems` to give
   * each child a known/stable CSS classname */
  hasOverflowScrollItems: PropTypes.bool,
  /** Whether to disable the default equal height cards */
  hasEqualColumnHeights: PropTypes.bool,
};

CardDeck.defaultProps = {
  className: undefined,
  columnSizes: {
    sm: 12,
    lg: 6,
    xl: 4,
  },
  hasInteractiveChildren: false,
  canScrollHorizontal: true,
  hasOverflowScrollItems: false,
  hasEqualColumnHeights: true,
};

CardDeck.Deprecated = BaseCardDeck;

export default CardDeck;
