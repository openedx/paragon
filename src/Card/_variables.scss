// Card

$card-spacer-y:                      .75rem !default;
$card-spacer-x:                      1.25rem !default;
$card-border-width:                  $border-width !default;
$card-border-radius:                 $border-radius !default;
$card-border-color:                  rgba($black, .125) !default;
$card-border-focus-color:            rgba($black, .5) !default;
$card-border-focus-color-dark:       theme-color("primary", "focus") !default;
$card-inner-border-radius:           subtract($card-border-radius, $card-border-width) !default;
$card-cap-bg:                        rgba($black, .03) !default;
$card-cap-color:                     null !default;
$card-height:                        null !default;
$card-color:                         null !default;
$card-bg:                            $white !default;
$card-bg-dark:                       $primary-500 !default;
$card-bg-muted:                      $light-200 !default;

$card-img-overlay-padding:           1.25rem !default;

$card-group-margin:                  calc($grid-gutter-width / 2) !default;
$card-deck-margin:                   $card-group-margin !default;
$card-grid-margin:                   $card-group-margin !default;
$card-deck-margin-bottom:            map_get($spacers, 3) !default;
$card-grid-margin-bottom:            map_get($spacers, 3) !default;

$card-columns-count:                 3 !default;
$card-columns-gap:                   1.25rem !default;
$card-columns-margin:                $card-spacer-y !default;

$card-divider-bg:                    $light-400 !default;
$card-divider-margin-y:              $card-spacer-y !default;

$card-footer-actions-gap:            .5rem !default;

$card-logo-left-offset:              1.5rem !default;
$card-logo-bottom-offset:            1rem !default;
$card-logo-left-offset-horizontal:   .4375rem !default;
$card-logo-bottom-offset-horizontal: .4375rem !default;

$card-logo-width:                    7.25rem !default;
$card-logo-height:                   4.125rem !default;

$card-image-border-radius:           .3125rem !default;
$card-logo-border-radius:            .25rem !default;

$card-footer-text-font-size:         $x-small-font-size;

$card-image-horizontal-max-width:    240px !default;
$card-image-horizontal-min-width:    $card-image-horizontal-max-width !default;
$card-image-vertical-max-height:     140px !default;
$loading-skeleton-spacer:            .313rem !default;

$card-focus-border-offset:           5px !default;
$card-focus-border-width:            2px !default;
$card-focus-border-radius:           5px !default;
