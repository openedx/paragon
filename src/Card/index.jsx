export { default } from 'react-bootstrap/Card';
export { default as CardColumns } from 'react-bootstrap/CardColumns';
export { default as CardDeck } from 'react-bootstrap/CardDeck';
export { default as CardImg } from 'react-bootstrap/CardImg';
export { default as CardGroup } from 'react-bootstrap/CardGroup';

export { default as CardGrid } from './CardGrid';
