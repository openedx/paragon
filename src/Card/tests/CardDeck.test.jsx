import React from 'react';
import renderer from 'react-test-renderer';
import { v4 as uuidv4 } from 'uuid';
import CardDeck from '../CardDeck';
import Card from '..';

function ExampleCard(props) {
  return (
    <Card {...props}>
      <Card.ImageCap src="http://fake.image" />
      <Card.Body>
        <Card.Header>Card title</Card.Header>
        <Card.Section>
          This is a wider card with supporting text below as a natural lead-in to
          additional content. This card has even longer content than the first to
          show that equal height action.
        </Card.Section>
      </Card.Body>
      <Card.Footer textElement={<small className="text-muted">Last updated 3 mins ago</small>} />
    </Card>
  );
}

function getCardContent({ cardCount = 5, ...cardProps } = {}) {
  return Array.from({ length: cardCount }).map(() => <ExampleCard key={uuidv4()} {...cardProps} />);
}

describe('<CardDeck />', () => {
  it('renders default columnSizes', () => {
    const cardContent = getCardContent();
    const tree = renderer.create((
      <CardDeck>
        {cardContent}
      </CardDeck>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with controlled columnSizes', () => {
    const cardContent = getCardContent();
    const tree = renderer.create((
      <CardDeck
        columnSizes={{
          xs: 12,
          md: 6,
          lg: 4,
          xl: 3,
        }}
      >
        {cardContent}
      </CardDeck>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has tabIndex="-1" when `hasInteractiveChildren` is true', () => {
    const cardContent = getCardContent({ isClickable: true });
    const tree = renderer.create((
      <CardDeck hasInteractiveChildren>
        {cardContent}
      </CardDeck>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with disabled equal height', () => {
    const cardContent = getCardContent();
    const tree = renderer.create((
      <CardDeck hasEqualColumnHeights={false}>
        {cardContent}
      </CardDeck>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
