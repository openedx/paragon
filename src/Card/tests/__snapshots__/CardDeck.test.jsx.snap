// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardDeck /> has tabIndex="-1" when \`hasInteractiveChildren\` is true 1`] = `
<div
  className="pgn__card-deck pgn__card-deck-has-horizontal-scroll"
>
  <div
    className="pgn__card-deck-row row"
    tabIndex={-1}
  >
    <div
      className="pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card clickable pgn__card-light card"
        tabIndex={0}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card clickable pgn__card-light card"
        tabIndex={0}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card clickable pgn__card-light card"
        tabIndex={0}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card clickable pgn__card-light card"
        tabIndex={0}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card clickable pgn__card-light card"
        tabIndex={0}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CardDeck /> renders default columnSizes 1`] = `
<div
  className="pgn__card-deck pgn__card-deck-has-horizontal-scroll"
>
  <div
    className="pgn__card-deck-row row"
    tabIndex={0}
  >
    <div
      className="pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CardDeck /> renders with controlled columnSizes 1`] = `
<div
  className="pgn__card-deck pgn__card-deck-has-horizontal-scroll"
>
  <div
    className="pgn__card-deck-row row"
    tabIndex={0}
  >
    <div
      className="pgn__card-deck__card-item col-xl-3 col-lg-4 col-md-6 col-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item col-xl-3 col-lg-4 col-md-6 col-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item col-xl-3 col-lg-4 col-md-6 col-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item col-xl-3 col-lg-4 col-md-6 col-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item col-xl-3 col-lg-4 col-md-6 col-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CardDeck /> renders with disabled equal height 1`] = `
<div
  className="pgn__card-deck pgn__card-deck-has-horizontal-scroll"
>
  <div
    className="pgn__card-deck-row row"
    tabIndex={0}
  >
    <div
      className="pgn__card-deck__card-item pgn__card__disable-equal-column-heights col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item pgn__card__disable-equal-column-heights col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item pgn__card__disable-equal-column-heights col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item pgn__card__disable-equal-column-heights col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck__card-item pgn__card__disable-equal-column-heights col-xl-4 col-lg-6 col-sm-12"
    >
      <div
        className="pgn__card pgn__card-light card"
        tabIndex={-1}
      >
        <div
          className="pgn__card-wrapper-image-cap vertical"
        >
          <img
            className="pgn__card-image-cap"
            onError={[Function]}
            onLoad={[Function]}
            src="http://fake.image"
          />
        </div>
        <div
          className="pgn__card-body"
        >
          <div
            className="pgn__card-header"
          >
            <div
              className="pgn__card-header-content"
            />
          </div>
          <div
            className="pgn__card-section"
          >
            This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
          </div>
        </div>
        <div
          className="pgn__card-footer vertical"
        >
          <div
            className="pgn__card-footer-text vertical"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
